new-frontiers-14/frontier-station-14
# Medical Bounties

## About the PR



* Adds medical bounties to the game.
* Removed the dungeon and salvage corpse spawners from the salvage chest spawners found on wrecks.
* All corpses found on wrecks now have names and medical bounties associated with them.
* All corpses found on wrecks and dungeons now have loadouts associated with them (edited versions of the visitor loadouts)

### TO DO:
* [X] Revise loadouts for corpse classes (removing unwanted equipment) (note: some still have basic sec armor, all weapons and fluff removed)
* [X] Add HideSpawnMenu to new medical bounty loadout entities
* [X] Add replacement sprites for the body retrieval unit
* [X] Fix UI layout
* [ ] Fix the slot contents shown in the right-click 
* [X] Map a body retrieval unit (probably in the back rooms of Frontier's medbay)
* [X] Add a tool to view the maximum and current bounty value of a medical bounty. (No current at the moment)

Upon spawning, any entity with a medical bounty receives damage according to a randomly selected preset.  The value of the bounty associated with them is assigned based on the total amount of received damage (and possible reagents added to the patient) - the idea here is to reward medics proportionally to the amount of work or materials put in to restoring the patient to health.

## Why / Balance


Medical bounties are intended to give medics something to do during downtime without being so profitable as to supplant rescuing other injured players.  This should reduce downtime as a medic for slower or more populated shifts, and provides a lower-stress option for experimenting with treatment.

## How to test


1. Fly around to wrecks.
2. Find one with a corpse (should be somewhat rare, expecting one per few wrecks, ideally a few minutes per find).  The corpse should be clothed.
3. Heal the damage on the body using the restore trick.
4. Spawn a body retrieval unit (MachineMedicalBountyRedemption).
5. Check the value of the bounty using VV (MedicalBountyComponent).
6. Load the corpse into the unit, use the UI, redeem the bounty, the full value of the bounty should spawn on top of the machine.
7. Spawn another corpse.
8. Open the UI for the unit, it should say nothing's loaded in.
9. Load in the body, it should say that the body has too much damage.
10. Heal most of the damage (or restore it and hit it a few times with a weapon), make sure the target is alive and not critical.
11. Load the body into the body retrieval unit.
12. The bounty should redeem upon pressing the button and a reduced bounty price should be available given the amount of damage the corpse has.

## Media


TBD.

## Requirements



## Breaking changes


**Changelog**


:cl:
- add: Medical bounties are available for all corpses found on wrecks.
